model/adjuster/anonymizer.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package adjuster
+
+import "github.com/uber/jaeger/model"
+
+// Anonymizer returns an adjuster that modifies the trace to remove any information that might
+// be considered private or confidential. Examples are service names, port numbers, operations,
+// and any user-added data.
+//
+// This allows real production traces to safely be kept in public repositories for use in test
+// suites, after anonymization.
+//
+// A custom mapping is provided that dictates how to anonymize data. If a value is encountered that
+// that not in the mapping, the mapping is modified to include the new value. Modifications are
+// deterministic.
+//
+// This adjuster never returns any errors.
+func Anonymizer(mapping AnonymizerMapping) Adjuster {
+	return Func(func(trace *model.Trace) (*model.Trace, error) {
+		return trace, nil
+	})
+}
+
+// AnonymizerMapping dictates how anonymization is done.
+//
+// It is a map from anonymized field name -> field value -> anonymized value
+//
+// An example of a mapping:
+//var mapping = AnonymizerMapping{
+//	adjuster.AnonymizedService: {
```

Comment:
Rather than providing a fixed mapping, another option is to provide a naming pattern like `"service-%03d"` that can be used to generate the replacement values (internally it can keep a map to ensure the same service name is mapped to the same replacement value). For even more flexibility, this can be a function that takes the original value and returns the replacement value.

Also, rather than passing this as a map, I would probably define a struct with public fields. In fact, that struct itself can be the anonymizer by implementing the Adjuster interface, e.g.
```go
a := &adjuster.Anonymizer{
    ServiceName: CountingStringMapper("service-%03d"),
}

func CountingStringMapper(pattern string) func(in string) string {
  values := make(map[string]string)
  return func(in string) string {
    if out, ok := values[in]; ok {
        return out
    }
    out := fmt.Sprintf(pattern, len(values))
    values[in] = out
    return out
  }
}

```

Note - if you go with this approach, you may want to move this to a sub-package. In fact, in order to even use this adjuster you need some way to retrieve the trace, so perhaps it should be a standalone util under `/cmd/anonymizer` that reads/writes JSON.